question-service/controller/question-controller.js

```diff
@@ -0,0 +1,26 @@
+import { ormGetRandomQuestion as _getRandomQuestion } from "../model/question-orm.js";
+
+export async function getRandomQuestion(req, res) {
+  try {
+    const { difficulty } = req.query;
+    console.log(req.query);
+    let type = req.query.type ? req.query.type : null;
+
+    if (difficulty) {
+      console.log(typeof difficulty);
+      const resp = await _getRandomQuestion(parseInt(difficulty), type);
+      console.log(resp);
+      if (resp.err) {
+        return res.status(400).json({ message: "Could not get question!" });
+      } else {
+        return res.status(201).json({ resp });
+      }
+    } else {
+      return res.status(400).json({ message: "Difficulty is missing!" });
+    }
+  } catch (err) {
+    return res
+      .status(500)
+      .json({ message: "Database failure when getting question!" });
+  }
+}
```

Comment:
Just a suggestion but I think writing in guard clauses might be better in terms of readability.
For example, the above could be written as:

```
export async function getRandomQuestion(req, res) {
  try {
    const { difficulty } = req.query;
    console.log(req.query);
    let type = req.query.type ? req.query.type : null;
    
    if (!difficulty) return res.status(400).json({ message: "Difficulty is missing!" });

    const resp = await _getRandomQuestion(parseInt(difficulty), type);
    
    if (resp.err) {
      return res.status(400).json({ message: "Could not get question!" });
    }

    return res.status(400).json({ message: "Difficulty is missing!" });
  } catch (err) {
    return res
        .status(500)
        .json({ message: "Database failure when getting question!" });
  }
}
```
I personally feel that it's more readable but feel free to keep it if you think otherwise! Also just be careful that the logic is equivalent too.